### ISSUE TITLE
iOS Crash CarouselView has MediaElement in ItemTemplate

### ISSUE THREAD

**Issue by Tframe** [dotnet/maui#14417]
### Description

When placing a MediaElement inside a CarouselView, the app will crash when ran on iOS, but not Android. See error below:

child view controller:<AVPlayerViewController: 0x7fc6f8244200> should have parent view controller:<Microsoft_Maui_Controls_Handlers_Items_CarouselViewController: 0x7fc6f092c390> but actual parent is:<Microsoft_Maui_Controls_Platform_Compatibility_ShellFlyoutRenderer: 0x7fc6d7f4e740>

### Steps to Reproduce

1. Create new .net Maui app with .net 7
2. Add CommunityToolkit.Maui.MediaElement (version 1.0.1 or 1.0.2) Nuget package
3. In MauiProgram.cs, add .UseMauiCommunityToolkitMediaElement()
`builder
            .UseMauiApp<App>()
            .UseMauiCommunityToolkitMediaElement()
            .ConfigureFonts(fonts =>
            {
                fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
            });`
4. In MainPage.xaml, replace ScrollView content with a MediaElement 
5. Confirm it runs correctly.
6. Replace the ScrollView with a CarouselView. Set a generic ItemSource array and set the MediaElement as the ItemTemplate. 
7. Confirm it crashes on iOS.

### Link to public reproduction project repository

https://github.com/Tframe/Maui-Projects

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 15, 16

### Did you find any workaround?

No like workarounds. 

### Relevant log output

```shell
child view controller:<AVPlayerViewController: 0x7fc6f8244200> should have parent view controller:<Microsoft_Maui_Controls_Handlers_Items_CarouselViewController: 0x7fc6f092c390> but actual parent is:<Microsoft_Maui_Controls_Platform_Compatibility_ShellFlyoutRenderer: 0x7fc6d7f4e740>
```

**Comment by samiracles** [comment]
Facing the same issue

**Comment by bwinklesky** [comment]
Bumping.  Having this issue with MediaElement in ItemTemplate

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 1.0. Can repro on iOS platform with sample project. Android works fine.
https://github.com/Tframe/Maui-Projects
![Screenshot 2023-06-12 161533](https://github.com/dotnet/maui/assets/77434865/39368fab-11a5-45ea-8bb8-f2467f034282)

**Comment by zafrkaya** [comment]
Hello @Tframe @samiracles @bwinklesky, I'm trying to develop an application similar to TikTok and Instagram Discover. The only solution I can think of is to display the medias in CarouselView. Were you able to find a different alternative for this problem?

**Comment by Tframe** [comment]
> Hello @Tframe @samiracles @bwinklesky, I'm trying to develop an application similar to TikTok and Instagram Discover. The only solution I can think of is to display the medias in CarouselView. Were you able to find a different alternative for this problem?

The only workaround I can come up with is for iOS to open the medias in a WebView. To get it to look decent I had to create a HtmlWebViewSource so I could style the WebView appropriately and then use EvaluateJavaScriptAsync to force pause/play.

**Comment by zafrkaya** [comment]
Thanks for response @Tframe, I wonder, did it satisfy you in terms of performance? It would be great to see an example of the application.

**Comment by donatellijl12** [comment]
@mattleibow  I don't believe this a duplicate. https://github.com/CommunityToolkit/Maui/issues/1214 is specific to opening a browser prior to presenting a media element. This issue is for a media element within a carousel view. It has nothing to do with opening a browser. A media element in a carousel view on iOS always causes a crash. In addition, the other issue apparently is fixed in .NET 8. This issue is not fixed by .NET 8.

**Comment by donatellijl12** [comment]
Submitted a PR here to fix this issue: https://github.com/CommunityToolkit/Maui/pull/1623
